internal/v5/archive_test.go

```diff
@@ -37,12 +40,36 @@ import (
 	"gopkg.in/juju/charmstore.v5-unstable/internal/v5"
 )
 
-type ArchiveSuite struct {
+type commonArchiveSuite struct {
```

Comment:
Do we actually need another ArchiveSuite (and hence commonArchiveSuite) ?

I'd suggest (if it works OK) just having a single ArchiveSuite - the main reason why this might not work ok is if any terms stuff needed to be set up before SetUpTest is called, but that doesn't appear to be the case AFAICS.